### PR TITLE
Adds documentation to headers (primarily)

### PR DESCRIPTION
**Pull request by cppcooper**

### CODE REVIEW

**Review by brianzhouzc — APPROVED** [review]
lgtm

File: tests/tests-filter.cpp [hunk 1]
```diff
@@ -23,28 +24,37 @@ TEST(filtering, store_filtered) {
...
-        std::cout << "Filtering.." << std::endl;
+        sprintf(msg,"%s file\n",(dicom.IsValid() ? "valid" : "invalid"));
+        DEBUG_LOG(1,msg);
+
         auto [filtered_buffer,filtered_size] = filter.ApplyFilter(dicom); //memcpy ran til end of file(buffer)
         DicomFile filtered_dicom(filtered_buffer.get(),filtered_size);
         ASSERT_TRUE(filtered_dicom.IsValid()); //first sign the memcpy didn't work
-        std::cout << "Filtered data: " << (filtered_dicom.IsValid() ? "valid" : "invalid") << std::endl;
+        sprintf(msg,"Filtering: %s\n", (filtered_dicom.IsValid() ? "passed" : "failed"));
+        DEBUG_LOG(1,msg);
```
Comment (cppcooper) [inline tests/tests-filter.cpp:51]: Replacing the print lines with the new debug log macro can allow us to disable certain log lines even in unit tests. Definitely useful when debugging a unit test.

File: include/core.h [hunk 2]
```diff
@@ -19,14 +19,25 @@ namespace globals {
...
+#define PLUGIN_ERRORS (-1)
+#define INFO 0
+#define VERBOSE_1 1
+#define VERBOSE_2 2
+
+#define LOGGING_LEVEL VERBOSE_1
 #ifndef NDEBUG
     #ifndef UNIT_TEST
-     #define DEBUG_LOG(msg) if(globals::context) OrthancPluginLogWarning(globals::context, msg);
+     #define DEBUG_LOG(L,msg) if(globals::context && L <= LOGGING_LEVEL) switch(L){ \
```
Comment (cppcooper) [inline include/core.h:31]: I do wonder if this should be moved to a new file `debug.h` but it's honestly not that big, so I figure why bother.

**Review by cppcooper — COMMENTED** [review]
These are probably the most important things

File: include/core.h (hunk 2 above)
Comment (iwanlevin) [reply include/core.h:31]: I agree if it gets bigger maybe we move it to its own file but for now, this should be fine

**Review by iwanlevin — DISMISSED** [review]
The changes look good to me I went over them all I agree with not adding a debug.h yet but maybe if that gets larger or before we release a final version we should include that. Having the print lines replaced with a debug log macro is also very useful!

**Review by iwanlevin — APPROVED** [review]

File: include/core.h [hunk 3]
```diff
@@ -19,14 +19,25 @@ namespace globals {
...
+        case PLUGIN_ERRORS: OrthancPluginLogError(globals::context, msg);break;     \
+        case INFO: OrthancPluginLogInfo(globals::context, msg); break;              \
+        case VERBOSE_1:                                                             \
+        case VERBOSE_2: OrthancPluginLogWarning(globals::context, msg); break;      \
+      }
```
[the remark this answers is not in the document]
Comment (brianzhouzc) [reply include/core.h:36]: What is `VERBOSE_1` used for?

File: include/dicom-element-view.h [hunk 4]
```diff
@@ -13,7 +15,7 @@
  *   - get hex for element id
  *   - get hex for entire tag (group+element)
  */
-class DicomElement {
+class DicomElementView {
     // below DicomElement locates and aliases data within the dicom's data buffer, then calculates appropriate length/size values
 public:
```
[the remark this answers is not in the document]
Comment (brianzhouzc) [reply include/dicom-element-view.h:20]: I agree, this is more intuitive.

File: include/core.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (cppcooper) [reply include/core.h:36]: 1 and 2 are basically just differently levels of detail 1 is less than 2
So the intent here is that if you've got levels of information that will ultimately spam the log, you probably want one of these. Think of them as log lines which regular users don't ever need or want to see.

File: include/core.h [hunk 5]
```diff
@@ -19,14 +19,25 @@ namespace globals {
     extern fs::perms file_permissions;
 }
 #endif
-#define UNIT_TEST
+
+#define PLUGIN_ERRORS (-1)
+#define INFO 0
+#define VERBOSE_1 1
+#define VERBOSE_2 2
```
[the remark this answers is not in the document]
Comment (brianzhouzc) [reply include/core.h:26]: I'm a bit confused with `VERBOSE_1` and `VERBOSE_2`. Does `VERBOSE_1` (log level 1) do anything?

[the remark this answers is not in the document]
Comment (brianzhouzc) [reply include/core.h:26]: Just saw your other comment, understood

[the remark this answers is not in the document]
Comment (cppcooper) [reply include/core.h:26]: okay great

File: src/dicom-file.cpp [hunk 6]
```diff
@@ -21,25 +21,25 @@ bool DicomFile::parse_file() {
     const char* readable_buffer = (const char*)data;
     size_t preamble = 128;
     size_t prefix = 4;
-    DEBUG_LOG("DicomFile: parsing dicom data");
+    DEBUG_LOG(1,"DicomFile: parsing dicom data");
     // there is no valid header if the file size is smaller than 132 bytes
     if(size <= preamble + prefix){
-        if(globals::context) OrthancPluginLogError(globals::context, "DicomFile does not have a valid size, cannot continue parsing");
```
[the remark this answers is not in the document]
Comment (brianzhouzc) [reply src/dicom-file.cpp:27]: I don't see any other either.

**Review by brianzhouzc — COMMENTED** [review]
Looks good. The comments are clear and the DEBUG macro will make logging info a lot easier.

**Review by brianzhouzc — APPROVED** [review]

**Pull request merged**